nle/nethack/nethack.py

```diff
@@ -93,34 +93,56 @@ def __init__(
         wizard=False,
         hackdir=HACKDIR,
     ):
+        global _nhinstances
         self._copy = copy
 
+        _nhinstances = _nhinstances + 1
+
         if not os.path.exists(hackdir) or not os.path.exists(
             os.path.join(hackdir, "sysconf")
         ):
             raise FileNotFoundError(
                 "Couldn't find NetHack installation at '%s'." % hackdir
             )
 
-        # Create a HACKDIR for us.
-        self._tempdir = tempfile.TemporaryDirectory(prefix="nle")
-        self._vardir = self._tempdir.name
-
-        # Save cwd and restore later. Currently libnethack changes
-        # directory on loading.
-        self._oldcwd = os.getcwd()
+        self.shared = False;
 
-        # Symlink a few files.
-        for fn in ["nhdat", "sysconf"]:
-            os.symlink(os.path.join(hackdir, fn), os.path.join(self._vardir, fn))
-        # Touch a few files.
-        for fn in ["perm", "logfile", "xlogfile"]:
-            os.close(os.open(os.path.join(self._vardir, fn), os.O_CREAT))
-        os.mkdir(os.path.join(self._vardir, "save"))
+        if _pynethack.supports_shared():
+          # "shared" mode does some hacky things to enable using a
+          # shared libnethack.so, prevents writing to any files, and does
+          # not chdir.
+          self.shared = True
+          dlpath = DLPATH
+          self._hackdir = hackdir
+        else:
 
-        # Hacky AF: Copy our so into this directory to load several copies ...
-        dlpath = os.path.join(self._vardir, "libnethack.so")
-        shutil.copyfile(DLPATH, dlpath)
+          # Create a HACKDIR for us.
+          if os.getenv("SLURM_JOBID"):
```

Comment:
This slurm stuff probably doesn't belong here and I'll most likely remove it from this PR